src/Symfony/Component/Translation/MessageCache.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Component\Config\ConfigCache;
+
+/**
+ * @author Abdellatif Ait Boudad <a.aitboudad@gmail.com>
+ */
+class MessageCache implements MessageCacheInterface
+{
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
+    /**
+     * @var bool
+     */
+    private $debug;
+
+    /**
+     * @param string $cacheDir
+     * @param bool   $debug
+     */
+    public function __construct($cacheDir, $debug = false)
+    {
+        $this->cacheDir = $cacheDir;
+        $this->debug = $debug;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isFresh($locale, array $options = array())
+    {
+        $currentResourcesHash = isset($options['resources_hash']) ? $options['resources_hash'] : '';
+        $cache = new ConfigCache($resource = $this->getResourcePath($locale), $this->debug);
+        $isFresh = $cache->isFresh();
+        if ($isFresh && $this->debug) {
+            list($catalogue, $resourcesHash) = include $resource;
+            if ($resourcesHash !== $currentResourcesHash) {
+                return false;
+            }
+        }
+
+        return $isFresh;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load($locale)
+    {
+        $resource = $this->getResourcePath($locale);
+        list($catalogue, $resourcesHash) = include $resource;
+
+        return $catalogue;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dump(MessageCatalogueInterface $messages, array $options = array())
+    {
+        $resourcesHash = isset($options['resources_hash']) ? $options['resources_hash'] : '';
+        // save file
+        $cache = new ConfigCache($this->getResourcePath($messages->getLocale()), $this->debug);
+        $fallbackContent = $this->getFallbackContent($messages);
+
+        $content = sprintf(<<<EOF
+<?php
+use Symfony\Component\Translation\MessageCatalogue;
+\$resourcesHash = '%s';
+\$catalogue = new MessageCatalogue('%s', %s);
+%s
+return array(\$catalogue, \$resourcesHash);
+EOF
+            ,
+            $resourcesHash,
+            $messages->getLocale(),
```

Comment:
I agree with you :), an end user can inject PHP code into messages !! but I'm not sure if [var_export](https://github.com/aitboudad/symfony/blob/translation_catalogue_cache/src/Symfony/Component/Translation/MessageCache.php#L91) solve the issue or not ? //cc @fabpot